waspc/src/Wasp/Generator/Common.hs

```diff
@@ -14,27 +14,16 @@ data ProjectRootDir
 -- | Range of node versions that node packages generated by this generator work correctly with.
 nodeVersionRange :: SV.Range
 nodeVersionRange =
-  SV.rangeFromVersionsIntersection
-    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
-      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
-    ]
+  SV.rangeFromVersion (SV.BackwardsCompatibleWith, latestLTSVersion)
   where
     latestLTSVersion = SV.Version 16 0 0
-    -- There is a bug in node 16.15.1 (more correctly, in npm 8.11.0 that comes with it)
-    -- that messes up how Wasp uses Prisma. That is why we limited ourselves to <=16.15.0 for now.
-    -- Bug issue on NPM CLI repo: https://github.com/npm/cli/issues/5018 .
-    latestLTSExactVersionThatWeKnowWorks = SV.Version 16 15 0
 
 -- | Range of npm versions that Wasp and generated projects work correctly with.
 npmVersionRange :: SV.Range
 npmVersionRange =
-  SV.rangeFromVersionsIntersection
-    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
-      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
-    ]
+  SV.rangeFromVersion (SV.BackwardsCompatibleWith, latestLTSVersion)
   where
-    latestLTSVersion = SV.Version 8 0 0 -- Goes with node 16
-    latestLTSExactVersionThatWeKnowWorks = SV.Version 8 5 5 -- Goes with node 16.15.0
+    latestLTSVersion = SV.Version 8 0 0 -- Goes with node 16 (but also higher versions too)
 
 prismaVersion :: SV.Version
```

Comment:
Instead of a range, just using `BackwardsCompatibleWith` for both. However, for NPM it is slightly harder to pin down since 8.x is used across all versions of node.